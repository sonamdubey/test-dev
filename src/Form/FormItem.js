import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class FormItem extends React.Component {
  static propTypes = {
    /** The content of the form item. */
    children: PropTypes.node,
    /** The helper message. */
    helperText: PropTypes.string,
    /** A prefix for `FormItem` component classes. */
    prefixClass: PropTypes.string,
    /** The validation status. */
    validateStatus: PropTypes.oneOf(['error', '']),
  };

  static defaultProps = {
    prefixClass: 'oxygen-form-item',
  };

  renderWrapper(child1, child2) {
    const { prefixClass, validateStatus } = this.props;

    let wrapperClassName = `${prefixClass}-control`;

    if (validateStatus) {
      wrapperClassName = classNames(wrapperClassName, {
        'has-error': validateStatus === 'error',
      });
    }

    return (
      <div className={wrapperClassName}>
        {child1}
        {child2}
      </div>
    );
  }

  renderHelper() {
    const { helperText, prefixClass } = this.props;

    if (helperText) {
      const helperTexterClassName = `${prefixClass}-help`;

      return <div className={helperTexterClassName}>{helperText}</div>;
    }

    return null;
  }

  render() {
    const { children, helperText, prefixClass } = this.props;

    let itemClassName = `${prefixClass}`;

    if (helperText) {
      itemClassName = classNames(itemClassName, [`${prefixClass}-with-help`]);
    }

    return (
      <div className={itemClassName}>
        {this.renderWrapper(children, this.renderHelper())}
      </div>
    );
  }
}

export default FormItem;
