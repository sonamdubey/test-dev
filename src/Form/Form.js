import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import FormItem from './FormItem';

class Form extends React.Component {
  static Item = FormItem;

  static propTypes = {
    /** A custom class for `form` element. */
    className: PropTypes.string,
    /** Callback fired when form is submitted. */
    onSubmit: PropTypes.func,
    /** A prefix for `Form` component classes. */
    prefixClass: PropTypes.string,
  };

  static defaultProps = {
    prefixClass: 'oxygen-form',
  };

  render() {
    const { className, prefixClass, ...otherProps } = this.props;

    const formClassName = classNames(prefixClass, className);

    return <form {...otherProps} className={formClassName} />;
  }
}

export default Form;
