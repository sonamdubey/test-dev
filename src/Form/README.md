# Form

## Usage
> Important: Make sure to include the [scss file](https://github.com/carwale/oxygen/blob/master/src/Form/style/form-item.scss) or feel free to create your own.
```js
import { Form, Input } from "oxygen"
// or
import Form from "oxygen/lib/Form"
import Input from "oxygen/lib/Input"
const FormItem = Form.Item

<Form>
  <FormItem
    validateStatus={this.state.usernameValidateStatus}
    helperText={this.state.usernamehelperText}
  >
    <Input
      id="name"
      label="Name"
      placeholder="Enter your name"
      value={this.state.username}
      onChange={this.handleUserNameChange}
    />
  </FormItem>
  <FormItem
    validateStatus={this.state.mobileValidateStatus}
    helperText={this.state.mobilehelperText}
  >
    <Input
      id="mobile"
      label="Mobile"
      type="tel"
      placeholder="Enter your mobile"
      value={this.state.mobile}
      onChange={this.handleMobileChange}
      prefix="+91"
      maxLength={10}
    />
  </FormItem>
  <FormItem>
    <button
      type="button"
      onClick={this.handleSubmit}
    >
      Submit
    </button>
  </FormItem>
</Form>
```

## API

### Form

#### Props

| propName | propType | defaultValue | isRequired | description |
| -------- | -------- | ------------ | ---------- | ----------- |
| className | `string` | - | - | A custom class for `form` element. |
| prefixClass | `string` | 'oxygen-form' | - | A prefix for `Form` component classes. |

#### Events

| propName | propType | defaultValue | isRequired | description |
| -------- | -------- | ------------ | ---------- | ----------- |
| onSubmit | `func` | - | - | Callback fired when form is submitted. |

### FormItem

#### Props

| propName | propType | defaultValue | isRequired | description |
| -------- | -------- | ------------ | ---------- | ----------- |
| helperText | `string` | - | - | The helper message. |
| prefixClass | `string` | 'oxygen-form-item' | - | A prefix for `FormItem` component classes. |
| validateStatus | oneOf `error`, `''` | - | - | The validation status. |
