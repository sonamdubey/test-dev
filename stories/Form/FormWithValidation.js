import React from 'react';

import Form from '../../src/Form';
import '../../src/Form/style/form-item.scss';
import Input from '../../src/Input';
import '../../src/Input/style/input.scss';
import Button from '../../src/Button';
import '../../src/Button/style/button.scss';

class FormWithValidation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      usernameValidateStatus: '',
      usernamehelperText: '',
      mobile: '',
      mobileValidateStatus: '',
      mobilehelperText: ''
    }
  }

  handleUserNameChange = (event) => {
    this.setState({
      username: event.target.value
    })
  }

  handleMobileChange = (event) => {
    this.setState({
      mobile: event.target.value
    })
  }

  handleSubmit = () => {
    if (this.state.username.length < 3) {
      this.setState({
        usernameValidateStatus: 'error',
        usernamehelperText: 'Enter name'
      })
    }
    else {
      this.setState({
        usernameValidateStatus: '',
        usernamehelperText: ''
      })
    }

    if (!this.state.mobile.length) {
      this.setState({
        mobileValidateStatus: 'error',
        mobilehelperText: 'Enter mobile number'
      })
    }
    else if (this.state.mobile.length < 10) {
      this.setState({
        mobileValidateStatus: 'error',
        mobilehelperText: 'Mobile number should be 10 digits'
      })
    }
    else {
      this.setState({
        mobileValidateStatus: '',
        mobilehelperText: ''
      })
    }
  }

  render() {
    return (
      <div className="example-container">
        <Form>
          <Form.Item
            validateStatus={this.state.usernameValidateStatus}
            helperText={this.state.usernamehelperText}
          >
            <Input
              id="name"
              label="Name"
              placeholder="Enter your name"
              value={this.state.username}
              onChange={this.handleUserNameChange}
            />
          </Form.Item>
          <Form.Item
            validateStatus={this.state.mobileValidateStatus}
            helperText={this.state.mobilehelperText}
          >
            <Input
              id="mobile"
              label="Mobile"
              type="tel"
              placeholder="Enter your mobile"
              value={this.state.mobile}
              onChange={this.handleMobileChange}
              prefix="+91"
              maxLength={10}
            />
          </Form.Item>
          <Form.Item>
            <Button type="primary" onClick={this.handleSubmit}>Submit</Button>
          </Form.Item>
        </Form>
      </div>
    )
  }
}

export default FormWithValidation;
