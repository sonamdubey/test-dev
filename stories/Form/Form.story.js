import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Form from '../../src/Form';
import '../../src/Form/style/form-item.scss';
import Input from '../../src/Input';
import '../../src/Input/style/input.scss';
import Button from '../../src/Button';
import '../../src/Button/style/button.scss';

import FormWithValidation from './FormWithValidation';

storiesOf('Form', module)
  .add('default',
    withInfo()(() => (
      <Form>
        <Form.Item>
          <Input
            id="name"
            label="Name"
            placeholder="Enter your name"
            value="John Doe"
          />
        </Form.Item>
        <Form.Item>
          <Input
            id="mobile"
            label="Mobile"
            type="tel"
            placeholder="Enter your mobile"
            value="9876543210"
            prefix="+91"
            maxLength={10}
          />
        </Form.Item>
        <Form.Item>
          <Button type="primary">Submit</Button>
        </Form.Item>
      </Form>
    ))
  )
  .add('with validation', () => (
    <FormWithValidation />
  ))
