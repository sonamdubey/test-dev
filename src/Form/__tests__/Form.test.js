import React from 'react';
import Form from '../index';

describe('<Form />', function() {
  describe('render', function() {
    it('should render a form', function() {
      const wrapper = shallow(
        <Form />
      )

      expect(wrapper).toMatchSnapshot();
    })
  })
})
