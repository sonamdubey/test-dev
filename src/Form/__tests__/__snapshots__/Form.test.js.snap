// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render should render a form 1`] = `
<form
  className="oxygen-form"
/>
`;
